src/main.c

```diff
@@ -528,6 +528,7 @@ int main(int argc, char * argv [])
 
       case 'D': /* disable auto erase */
         uflags &= ~UF_AUTO_ERASE;
+        avr_mem_hiaddr(NULL); /* disable trailing 0xff optimisation */
```

Comment:
`avr_mem_hiaddr(NULL)` looks strange. How about creating a new API for this purpose? May I suggest something like
`disable_trailing_ff_optimisation()`? That way we could remove the comment.

Reply:
Not a prob. Can do so almost trivially by 
```
#define disable_trailing_ff_optimisation() avr_mem_hiaddr(NULL)
```
in src/libavrdude.h in the line preceding the declaration of `int avr_mem_hiaddr(AVRMEM * mem);`

This (in my world) passes as a self-documenting API. OK?